### PR TITLE
`realpath` polyfill for shell

### PR DESCRIPTION
**Pull request by barelyhuman**
- Downloads fail with 401 due to the typo
- realpath doesn't exist on osx by default 

Both commits are atomic can be picked based on what you want to go with based on this issue 

PR based on this https://github.com/egoist/bina/issues/3

### CODE REVIEW

**Comment by egoist** [comment]
which repo fails with 401 error?

**Comment by barelyhuman** [comment]
Anything I tried failed, that's when I changed it to `browser_download_url`

image

**Comment by egoist** [comment]
does this work for you?

```
curl -SL -H "accept:application/octet-stream" -o foo https://api.github.com/repos/denoland/deno/releases/assets/50945986
```

**Comment by barelyhuman** [comment]
It does, yes.


Screenshot 2021-12-04 at 12 33 03 PM

**Comment by egoist** [comment]
change back to `browser_download_url` will make private repos stop working

**Comment by barelyhuman** [comment]
makes sense, I'm looking for what's causing it, it's probably the header being incomplete

File: make_sh.ts [hunk 1]
```diff
@@ -191,7 +191,7 @@ export const makeInstallerScript = ({
       ${platforms
         .map((p) => {
           return `${p.platform}) 
-          download_url="${p.asset.url}" 
+          download_url="${p.asset.browser_download_url}"
```
Comment (egoist) [inline make_sh.ts]: can you revert this so I can merge the realpath fix first

Comment (barelyhuman) [reply make_sh.ts]: Done

**Comment by barelyhuman** [comment]
let me test the preview urls first, **do not merge** right now

**Comment by barelyhuman** [comment]
~~the polyfill won't work with `sh`, works fine with `bash` , looking for an alternative~~

Edit: My bad, forgot to change the invocation syntax

**Comment by barelyhuman** [comment]
Found it, it's the `Authorization` header, since it has spaces after `token` and the whole this is added without being considered as a string, it's failing with 401. 

Changed
```
)} -w '%{http_code}' -H "$header1" -H "$header2" -H "$header3" -sSL -o "$local_file" "$source_url")
```
to this 
```
)} -w '%{http_code}' -H "$header1" -H '"$header2"' -H "$header3" -sSL -o "$local_file" "$source_url")
```

basically added `'` around the `$header2` to force it to be a string, hacky, I know. but I need confirmation if it works fine on linux with the forced string around, docs say it'll work , I just need a 2nd confirmation.

Want me to push this in the same PR?

**Comment by egoist** [comment]
Let's start another PR, I guess this is ok to merge now?

**Comment by barelyhuman** [comment]
Yeah you can merge this,  works on sh,bash, zsh fine, 
if you'd like to test it on other linux env, feel free, maybe add a github action for the same

**Pull request merged**